supabase/functions/admin-create-user/index.ts

```diff
@@ -0,0 +1,79 @@
+// @ts-nocheck (for Deno references, annoyingly cannot get around for now)
+/* global Deno */
+
+import { type Handler, serve } from "https://deno.land/std@0.168.0/http/server.ts";
+import { createClient } from "https://esm.sh/@supabase/supabase-js@latest";
+import { corsHeaders } from "../_shared/cors.ts";
+
+interface CreateUserRequestBody {
+    email: string;
+    password: string;
+    role: "admin" | "caller";
+}
+
+serve(async (req: Handler): Promise<Response> => {
+    if (req.method === "OPTIONS") {
+        return new Response("ok", { headers: corsHeaders });
+    }
+
+    const generateHeaders = (status: number): any => ({
+        ...corsHeaders,
+        "Content-Type": "application/json",
+        status,
+    });
+
+    const SUPABASE_URL = Deno.env.get("SUPABASE_URL");
+    const SUPABASE_ANON_KEY = Deno.env.get("SUPABASE_SERVICE_ROLE_KEY");
+    const supabase = createClient(SUPABASE_URL, SUPABASE_ANON_KEY, {
+        global: {
+            headers: {
+                Authorization: req.headers.get("Authorization")!,
+            },
+        },
+    });
+
+    // -- Prevent Non-Admin Access -- //
```

Comment:
did you say you were going to generalise this bit to the _shared folder?

Reply:
I've added it to my `TODO` list and I'll change it in a separate PR